pyne/openmc_utils.py

```diff
@@ -156,3 +171,274 @@ def xml(self):
+    rel_err = np.zeros_like(flux.std_dev)
+    nonzero = flux.mean > 0
+    rel_err[nonzero] = flux.std_dev[nonzero] / flux.mean[nonzero]
+    rel_err = np.reshape(rel_err.flatten(), newshape=(num_e_groups, num_ves))
+    rel_err = rel_err.transpose()
+    # calculate rel_err_tot, assuming independent in each energy bin (not true)
+    # due to the lack of covariance, this could be smaller than the true value
+    rel_err_tot = np.zeros_like(res_tot)
+    std_dev = np.reshape(flux.std_dev.flatten(),
+                         newshape=(num_e_groups, num_ves))
+    std_dev = std_dev.transpose()
+    var_tot = np.sum(np.square(std_dev), axis=1)
+    nonzero = res_tot > 0
+    rel_err_tot[nonzero] = np.sqrt(var_tot[nonzero]) / res_tot[nonzero]
+
+    # get volume averged results
+    result = np.divide(result, ve_vol)
+    res_tot = np.divide(res_tot, ve_vol)
+    # In the result of OpenMC, x changes fastest
+    # change the order to z changes fastest
+    result = result_changes_order(result, m.dims[3:6])
+    rel_err = result_changes_order(rel_err, m.dims[3:6])
+    res_tot = result_changes_order(res_tot, m.dims[3:6])
+    rel_err_tot = result_changes_order(rel_err_tot, m.dims[3:6])
+    return result, rel_err, res_tot, rel_err_tot
+
+def result_changes_order(result, dims):
+    """
+    This function changes the order of openmc flux data. The default order of
+    OpenMC flux is x changes the fastest. This function changes it to the order
+    which z changes the fasest (the same order as MCNP meshtal).
+
+    Parameters:
+    -----------
+    result : numpy array
+        This numpy array contains the flux data read from OpenMC state point
+        file. The shape of this numpy array is (num_ves*num_e_groups) or
+        (num_ves, ), with x changes the fastest.
+    dims : list
+        The dimension of x, y, z.
+
+    Returns:
+    --------
+    result_ : numpy array
+        This numpy array contains the flux data read from OpenMC state point
+        file. The shape of this numpy array is (num_ves*num_e_groups) or
+        (num_ves, ), with changes the fastest.
+    """
+
+    x_dim, y_dim, z_dim = dims[0], dims[1], dims[2]
+    num_e_groups = len(result.flatten()) // (x_dim * y_dim * z_dim)
+    result_ = np.reshape(result, newshape=(z_dim, y_dim, x_dim, num_e_groups))
+    # set energy changes fastest, z next, y next and then x
+    result_ = np.moveaxis(result_, [0, 1, 2, 3], [2, 1, 0, 3])
+    result_ = np.reshape(result_, newshape=(x_dim*y_dim*z_dim, num_e_groups))
+    return result_
+
+
+def create_meshtally(filename, tally_id, mesh_id=None, particle=None,
+                     tag_names=None, mesh_has_mats=False):
+    """
+    This function creates a MeshTally instance from OpenMC statepoint file.
+
+    Parameters:
+    -----------
+    filename : str
+        Filename of the OpenMC statepoint file. It ends with ".h5",
+        eg: "statepoint.10.h5".
+    tally_id : int
+        Tally number.
+    mesh_id : int
```

Comment:
Nice catch. In the very beginning version of this work, I used `mesh_id` to find out the mesh because OpenMC Python API is not yet an optional dependency (I can't differentiate mesh filter and energy filter) .

But later on, when OpenMC Python API was added as an optional dependency, I sued the filter check to find out the corresponding mesh filter for a given tally. Therefore, the `mesh_id` is not required now. But I forgot to remove it.

Thanks for notice this, and the `mesh_id` is remove in https://github.com/pyne/pyne/pull/1202/commits/5670c4d7e6fb1dc4a7b5af0339a41045b07e6be4.